lib/ibm_cloud_sdk_core/api_exception.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require("json")
+# Custom exception class for errors returned from the APIs
+class ApiException < StandardError
+  attr_reader :code, :error, :info, :transaction_id, :global_transaction_id
+  # :param HTTP::Response response: The response object from the API
+  def initialize(code: nil, error: nil, info: nil, transaction_id: nil, global_transaction_id: nil, response: nil)
+    if code.nil? || error.nil?
+      @code = response.code
+      @error = response.reason
+      unless response.body.empty?
+        body_hash = JSON.parse(response.body.to_s)
+        @code = body_hash["code"] || body_hash["error_code"] || body_hash["status"]
+        @error = body_hash["error"] || body_hash["error_message"] || body_hash["statusInfo"] || body_hash["description"]
+        %w[code error_code status error error_message statusInfo description].each { |k| body_hash.delete(k) }
+        @info = body_hash
```

Comment:
Can users get all the response headers?

Reply:
I have updated the ordering as mentioned by Phil, and we are removing the keys we use in `@code` or `@error` and rest of the `body_hash` is returned.